### DISCUSSION TITLE
iOS Pag UITableViewCell reuse question

### DISCUSSION THREAD

**Discussion by AppHock** [Tencent/libpag#1903] [Q&A] [answered]
【复现场景】UITableView reload的时候，通过cellForRowAt 更新 cell 的 pagview 的路径，pagView 未刷新页面，导致 cell 复用，展示之前旧的帧，和当前 cell 的 pag 资源对不上，必须要手动 flush 一次才行。

【需求背景】因为我们默认是暂停 pag 播放，只有手动滚动后才触发播放效果

【版本 】目前用的版本是4.0.5.60，只有再 iOS16 以上的机型上必现这个复用pag 和暂停的封面图对不上。

【问题】为啥设置了 path 你们没有去刷新当前帧，接入方很难猜到需要 flush 一次，幸好我做过 opengl 绘制，着色器绘制前会 clear 当前帧缓冲区颜色
`if item.isPag {
            guard let pagPath = item.pagLocalURL?.path, FileManager.default.fileExists(atPath: pagPath) else {
                return
            }
            
            self.pagFile = PAGFile.load(pagPath)
            self.pagView.setComposition(self.pagFile)
            self.pagView.flush()
            self.shouldAnimation ? self.pagView.play() : self.pagView.stop()
            self.staticImageView.isHidden = true
            self.pagView.isHidden = false
        }`

**Comment by kevingpqi123** [comment, marked as answer]
这类问题建议看下 API，setPath 或 setComposition 仅仅代表数据源发生了变化，又没有触发播放操作和 flush，为啥要刷新数据，API 注释中从来没有标注过会刷新当前帧
而在 flush 接口中清晰的注明了：Call this method to render current position immediately.

**Reply by AppHock** [reply]
那是，但是别人用SDK，肯定是希望能使用起来简单些，flush 这个方式备注我看了，但是别人很难想到 cell 复用的时候，pagView 还是显示旧那一个帧，只能猜测是不是要取主动 flush 一次。
